src/renderer/dx/DxRenderer.cpp

```diff
@@ -12,6 +12,177 @@
+    { 0.00038771, 0.01330373, 0.11098164, 0.22508352, 0.11098164, 0.01330373, 0.00038771 },
+    { 0.00019117, 0.00655965, 0.05472157, 0.11098164, 0.05472157, 0.00655965, 0.00019117 },
+    { 0.00002292, 0.00078633, 0.00655965, 0.01330373, 0.00655965, 0.00078633, 0.00002292 },
+    { 0.00000067, 0.00002292, 0.00019117, 0.00038771, 0.00019117, 0.00002292, 0.00000067 }
+};
+
+float4 Blur(Texture2D input, float2 tex_coord)
+{
+    uint width, height;
+    shaderTexture.GetDimensions(width, height);
+
+    float textureWidth = 1.0f/width;
+    float textureHeight = 1.0f/height;
+
+    float4 color = { 0, 0, 0, 0 };
+    float factor = 1;
+
+    int start = 0, end = 7; // sizeof(gaussianKernel[0])
+    for (int x = start; x < end; x++) 
+    {
+        float2 samplePos = { 0, 0 };
+
+        samplePos.x = tex_coord.x + (x - 7/2) * textureWidth;
+        for (int y = start; y < end; y++)
+        {
+            samplePos.y = tex_coord.y + (y - 7/2) * textureHeight;
+            color += input.Sample(samplerState, samplePos) * gaussianKernel[x][y] * factor;
+        }
+    }
+
+    return color;
+}
+
+float Gaussian2D(float x, float y, float sigma)
+{
+    return 1/(sigma*sqrt(2*M_PI)) * exp(-0.5*(x*x + y*y)/sigma/sigma);
+}
+
+float4 Blur2(Texture2D input, float2 tex_coord, float sigma)
+{
+    uint width, height;
+    shaderTexture.GetDimensions(width, height);
+
+    float texelWidth = 1.0f/width;
+    float texelHeight = 1.0f/height;
+
+    float4 color = { 0, 0, 0, 0 };
+    float factor = 1;
+
+    int sampleCount = 21;
+
+    for (int x = 0; x < sampleCount; x++) 
+    {
+        float2 samplePos = { 0, 0 };
+
+        samplePos.x = tex_coord.x + (x - sampleCount/2) * texelWidth;
+        for (int y = 0; y < sampleCount; y++)
+        {
+            samplePos.y = tex_coord.y + (y - sampleCount/2) * texelHeight;
+            if (samplePos.x <= 0 || samplePos.y <= 0 || samplePos.x >= width || samplePos.y >= height) continue;
+
+            color += input.Sample(samplerState, samplePos) * Gaussian2D((x - sampleCount/2), (y - sampleCount/2), sigma);
+        }
+    }
+
+    return color;
+
+}
+
+float SquareWave(float y)
+{
+    // Square wave gogogo.
+    // return 1.0 - SCANLINE_FACTOR * (1 - fmod(pos.y, 2.0));
+    // return sin(pos.y / SCANLINE_PERIOD * 2.0 * M_PI) >= 0.0 ? 1: SCANLINE_FACTOR;
+    return 1 - (floor(y / SCANLINE_PERIOD) % 2) * SCANLINE_FACTOR;
+}
+
+float4 Scanline(float4 color, float4 pos)
+{
+    float wave = SquareWave(pos.y);
+
+    if (length(color.rgb) < 0.2 && false)
+    {
+        return color + wave*0.1;
+    }
+    else
+    {
+        return color * wave;
+    }
+}
+
+float4 main(float4 pos : SV_POSITION, float2 tex : TEXCOORD) : SV_TARGET
+{
+    // return float4(0, 0.5f, 0.5f, 1);
+    // return shaderTexture.Sample(SampleType, tex) * float4(1.0, 0.5, 0.5, 1.0);
+    // return float4( 1.0f, 1.0f, 0.0f, 1.0f );    // Yellow, with Alpha = 1
+    // return color;
+    float4 greener = float4(0.5, 2, 0.5, 1);

```

Comment:
I really like this idea, but let's file a follow-up issue to track adding it as a setting :)